File: how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.cpp
class Solution {
public:
  vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
    vector < int > result;
    
    map < int, int > mp;
    for ( int i = 0; i < nums.size(); i++ ) {
      int count = 0;
      if ( mp[nums[i]] == 0 ) {
        for ( int j = 0; j < nums.size(); j++ ) {
          if ( i != j && nums[i] > nums[j] ) {
            count++;
          }
        }

        result.push_back(count);
        mp[nums[i]] = count;
      } else {
        result.push_back(mp[nums[i]]);
      }
    }

    return result;
  }
};

File: how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function(nums) {
  let result = []

  let mp = new Map()
  for ( let i = 0; i < nums.length; i++ ) {
    let count = 0
    if (mp.get(nums[i]) == undefined ) {
      for ( let j = 0; j < nums.length; j++ ) {
        if ( i != j && nums[i] > nums[j] ) {
          count++;
        }
      }
      
      result.push(count)
      mp.set(nums[i], count)
    } else {
      result.push(mp.get(nums[i]))
    }
  }

  return result
};